sql/src/main/java/org/apache/druid/sql/calcite/schema/MetadataSegmentView.java

```diff
@@ -143,10 +150,16 @@ private void poll()
     while (metadataSegments.hasNext()) {
       final SegmentStatusInCluster segment = metadataSegments.next();
       final DataSegment interned = DataSegmentInterner.intern(segment.getDataSegment());
+      Integer replicationFactor = segment.getReplicationFactor();
+      if (segment.getReplicationFactor() == null) {
+        replicationFactor = segmentIdToReplicationFactor.getIfPresent(segment.getDataSegment().getId());
+      } else {
+        segmentIdToReplicationFactor.put(segment.getDataSegment().getId(), segment.getReplicationFactor());
+      }
       final SegmentStatusInCluster segmentStatusInCluster = new SegmentStatusInCluster(
           interned,
           segment.isOvershadowed(),
-          segment.getReplicationFactor()
+          replicationFactor
```

Comment:
replicationFactor still can be null rite ?

Reply:
If the coordinator update and the cache both do not have any values for it, it would be null, and -1 is displayed in the segments table. I think allowing null values here is okay in this case. This should be the behaviour, right?

Reply:
Yeah if we donot have information in the cache and that information is not available in the coordinator, it should be displayed as -1.